app/assets/v2/js/notifications.js

```diff
@@ -0,0 +1,269 @@
+    hasNext,
+    numPages,
+    numNotifications
+  },
+  mounted() {
+    this.fetchNotifications();
+  },
+  created() {
+    this.sendState();
+  }
+});
+
+if (document.getElementById('gc-inbox')) {
+  var appInbox = new Vue({
+    delimiters: [ '[[', ']]' ],
+    el: '#gc-inbox',
+    data() {
+      return {
+        page,
+        notifications,
+        unreadNotifications,
+        hasNext,
+        numPages,
+        numNotifications,
+        selectedNotifications: []
+      };
+    },
+    computed: {
+      selectAll: {
+        get: function() {
+          return this.notifications ? this.selectedNotifications.length == this.notifications.length : false;
+        },
+        set: function(value) {
+          var selectedNotifications = [];
+
+          if (value) {
+            this.notifications.forEach(function(notification) {
+              selectedNotifications.push(notification.id);
+            });
+          }
+
+          this.selectedNotifications = selectedNotifications;
+        }
+      }
+    },
+    methods: {
+      toggleRead(unread) {
+        vm = this;
+
+        vm.notifications.map((notify, index) => {
+          if (vm.selectedNotifications.includes(notify.id))
+            if (unread) {
+              notify.is_read = false;
+            } else {
+              notify.is_read = true;
+            }
+        });
+        window.sessionStorage.setItem('notificationRead', vm.selectedNotifications);
+        if (unread) {
+          vm.sendState(true);
+        } else {
+          vm.sendState();
+        }
+      },
+      deleteNotification() {
+        vm = this;
+        const deleteObj = Object();
+
+        deleteObj['delete'] = vm.selectedNotifications;
+        let data = JSON.stringify(deleteObj);
+        let deleteNotify = fetchData ('/inbox/notifications/delete/', 'DELETE', data);
+
+        $.when(deleteNotify).then(function(response, status, statusText) {
+
+          if (statusText.status === 204) {
+            vm.notifications.map((notify, index) => {
+              if (vm.selectedNotifications.includes(notify.id))
+                vm.notifications.splice(notify[index], 1);
+            });
+            vm.selectedNotifications = [];
+          }
+        });
+      },
+      selectUnread() {
+        vm = this;
+        vm.selectedNotifications = [];
+        vm.notifications.forEach(function(notification) {
+
+          if (notification.is_read === false) {
+            vm.selectedNotifications.push(notification.id);
+          }
+        });
+      },
+      selectRead() {
+        vm = this;
+        vm.selectedNotifications = [];
+        vm.notifications.forEach(function(notification) {
+
+          if (notification.is_read === true) {
+            vm.selectedNotifications.push(notification.id);
+          }
+        });
+      }
+    },
+    mounted() {
+      this.fetchNotifications();
+    },
+    created() {
+      this.sendState();
+    }
+  });
+}
+
+function newData(newObj, oldObj) {
```

Comment:
could we shift to using function variables ? 
` const newDate = (newObj, oldObj) => { };`